<a href="https://colab.research.google.com/github/Shiromasteven/Project-2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2
##Steven Shiroma
###Date7/7/2022

In [48]:
import pandas as pd

filename1 = ('https://raw.githubusercontent.com/fivethirtyeight/data/da9a637f0e2fa735f2c0d9e733c2e971e7563514/bad-drivers/bad-drivers.csv')
filename2 = ('/content/drive/MyDrive/Stars.csv')

df_drivers = pd.read_csv(filename1)
df_stars = pd.read_csv(filename2)

df_drivers.head()

State  \
0     Alabama   
1      Alaska   
2     Arizona   
3    Arkansas   
4  California   

   Number of drivers involved in fatal collisions per billion miles  \
0                                               18.8                  
1                                               18.1                  
2                                               18.6                  
3                                               22.4                  
4                                               12.0                  

   Percentage Of Drivers Involved In Fatal Collisions Who Were Speeding  \
0                                                 39                      
1                                                 41                      
2                                                 35                      
3                                                 18                      
4                                                 35                      

   Percentage Of Drivers Involved In Fatal Collisions Who Were Alcohol-Impaired  \
0                                                 30                              
1                                                 25                              
2                                                 28                              
3                                                 26                              
4                                                 28                              

   Percentage Of Drivers Involved In Fatal Collisions Who Were Not Distracted  \
0                                                 96                            
1                                                 90                            
2                                                 84                            
3                                                 94                            
4                                                 91                            

   Percentage Of Drivers Involved In Fatal Collisions Who Had Not Been Involved In Any Previous Accidents  \
0                                                 80                                                        
1                                                 94                                                        
2                                                 96                                                        
3                                                 95                                                        
4                                                 89                                                        

   Car Insurance Premiums ($)  \
0                      784.55   
1                     1053.48   
2                      899.47   
3                      827.34   
4                      878.41   

   Losses incurred by insurance companies for collisions per insured driver ($)  
0                                             145.08                             
1                                             133.93                             
2                                             110.35                             
3                                             142.39                             
4                                             165.63

In [49]:
df_drivers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51 entries, 0 to 50
Data columns (total 8 columns):
 #   Column                                                                                                  Non-Null Count  Dtype  
---  ------                                                                                                  --------------  -----  
 0   State                                                                                                   51 non-null     object 
 1   Number of drivers involved in fatal collisions per billion miles                                        51 non-null     float64
 2   Percentage Of Drivers Involved In Fatal Collisions Who Were Speeding                                    51 non-null     int64  
 3   Percentage Of Drivers Involved In Fatal Collisions Who Were Alcohol-Impaired                            51 non-null     int64  
 4   Percentage Of Drivers Involved In Fatal Collisions Who Were Not Distracted                     

1) Source of data
https://github.com/fivethirtyeight/data/blob/da9a637f0e2fa735f2c0d9e733c2e971e7563514/bad-drivers/bad-drivers.csv

as well as the article here:

https://fivethirtyeight.com/features/which-state-has-the-worst-drivers/

2) Brief description of data

This is data across 51 states of accidents, fatalities, and insurance data.

3) What is the target?

Car insurance premiums, the question being how much can individuals, and companies expect insurance premiums to be based on state driver information.

4) Is this a classification or regression problem?

This is a regression problem


5) How many features?

7

6) How many rows of data.

51


7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

many integers that are actually percents and need to be cast into floats.


In [50]:
df_stars.head()

Temperature         L       R    A_M Color Spectral_Class  Type
0         3068  0.002400  0.1700  16.12   Red              M     0
1         3042  0.000500  0.1542  16.60   Red              M     0
2         2600  0.000300  0.1020  18.70   Red              M     0
3         2800  0.000200  0.1600  16.65   Red              M     0
4         1939  0.000138  0.1030  20.06   Red              M     0

In [54]:
df_stars.shape

(240, 7)

Second choice: dataset 2

1) Source of data

https://www.kaggle.com/datasets/brsdincer/star-type-classification


2) Brief description of data

Statistics abount different characteristics of stars

3) What is the target?

Type

4) Is this a classification or regression problem?

classification

5) How many features?

7

6) How many rows of data.

240

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

whether or not to ohe or ordinal encoe the Smass/ hertzsprung-russell classification of stars